versioned_docs/version-2.0/how_to_guides/evaluation/upload_existing_experiments.mdx

```diff
@@ -0,0 +1,268 @@
+---
+sidebar_position: 13
+---
+
+import {
+  CodeTabs,
+  PythonBlock,
+  TypeScriptBlock,
+} from "@site/src/components/InstructionsWithCode";
+import { RegionalUrl } from "@site/src/components/RegionalUrls";
+
+# Upload experiments run outside of LangSmith with the REST API
+
+Some users prefer to manage their datasets and run their experiments outside of LangSmith, but want to use the LangSmith UI to view the results. This is supported via our `/datasets/upload-experiment` endpoint.
+
+This guide will show you how to upload evals using the REST API, using the `requests` library in Python as an example. However, the same principles apply to any language.
+
+## Request body schema
+
+Uploading an experiment requires specifying the relevant high-level information for your experiment and dataset, along with the individual data for your examples and runs within
+the experiment. Each object in the `results` represents a "row" in the experiment - a single dataset example, along with an associated run. You may use the
+following schema to upload experiments to the `/datasets/upload-experiment` endpoint:
+
+```json
+{
+  "experiment_name": "string (required)",
+  "experiment_description": "string (optional)",
+  "experiment_start_time": "datetime (required)",
+  "experiment_end_time": "datetime (required)",
+  "dataset_id": "uuid (optional - must provide either dataset_id or dataset_name)",
+  "dataset_name": "string (optional - must provide either dataset_id or dataset_name)",
```

Comment:
important to mention that this refers to a dataset id in the external system -- used to group experiments together

Reply:
otherwise users might be confused and think that they need an id in langsmith

Reply:
Got it, yeah - but just to be clear, we do set this as the dataset_id in langsmith (but i do see what you're saying)